.github/workflows/integration_test.yaml

```diff
@@ -8,18 +8,17 @@ jobs:
     name: integration tests
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v4
-    - name: Set up Python 
-      uses: actions/setup-python@v4
-    - run: sudo snap install charmcraft --channel=2.x/stable --classic
-    - run: sudo snap install juju --channel=3.3/stable
-    - uses: charmed-kubernetes/actions-operator@main 
+    - name: checkout
+      uses: actions/checkout@v4
+    # See: https://github.com/charmed-kubernetes/actions-operator/issues/82
+    - name: Setup Python
+      uses: actions/setup-python@v5
+      with:
+        python-version: "3.12"
+    - name: setup Juju controller and dependencies
+      uses: charmed-kubernetes/actions-operator@main 
       with: 
-        juju-channel: 3.3/stable 
+        juju-channel: 3.5/stable 
```

Comment:
question: is there a minimum juju version we suggest in our docs for the on-prem livepatch server? if we do, then we should be testing with that version of juju.. or possibly test with an array of juju versions to make sure nothing breaks..

Reply:
There isn't a minimum Juju version suggested. That would be good to add. This minimum requirement on 3.5 is because of the Postgresql charm.